examples/chat-app/main.go

```diff
@@ -144,21 +166,35 @@ func main() {
 		}
 
 		// Initialize client config
-		env := &clientConfig{
+		client := &clientConfig{
 			ctx:      ctx,
 			client:   smClient,
 			room:     *room,
 			serverID: info.ID,
 			nick:     *nick,
 			self:     host.ID(),
 		}
-
-		// Initialize chat UI
-		ui := newUI(env)
-		// Run UI, sync existing messages and start prompt
-		err = ui.start()
-		if err != nil {
-			panic("Error initializing UI")
+		// an input chan for typing messages into
+		outCh := make(chan string, 32)
+		// Start listening to STDIN for new messages
+		go client.readInput(outCh)
+		// Print my messages and every new message in STDOUT
+		go client.writeOutput(outCh)
+		// Initial sync in case there are old messages in room
+		client.syncMessages(outCh)
+
+		// Periodically listen for new messages.
+		msgSyncTicker := time.NewTicker(syncTime)
+		defer msgSyncTicker.Stop()
+		for {
+			select {
+			case <-msgSyncTicker.C:
+				// We periodically fetch the smrat-record to check if there are new messages.
+				client.syncMessages(outCh)
```

Comment:
Same comment as in the GC PR: The timer should be reset each time after the completion of syncMessages